### PR TITLE
Fix sidecar not exit when job completes

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

This handle quits forcibly as this is called from application container, we donot need to gracefully terminate, not to mention the graceful termination can not work now.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@ramaraochavali

File: pilot/cmd/pilot-agent/status/server.go [hunk 1]
```diff
@@ -603,8 +603,8 @@ func (s *Server) handleQuit(w http.ResponseWriter, r *http.Request) {
 	}
 	w.WriteHeader(http.StatusOK)
 	_, _ = w.Write([]byte("OK"))
-	log.Infof("handling %s, notifying pilot-agent to exit", quitPath)
-	notifyExit()
+	log.Infof("handling %s, exiting...", quitPath)
+	go os.Exit(0)
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/status/server.go:607]: I need to start a new routine, otherwise the handler will have no chance to return

**Comment by ramaraochavali** [comment]
> we donot need to gracefully terminate, not to mention the graceful termination can not work now.

But /quitquitquit can be called for other use cases where we need graceful termination and it works now. But I agree that graceful termination is not needed for jobs but we can distinguish that is job which called it?

**Comment by hzxuzhonghu** [comment]
For long-running pod, it will not call /quitquitquit

**Comment by ramaraochavali** [comment]
So if it will not call /quitquitquit when will this code gets executed?

**Comment by hzxuzhonghu** [comment]
The long running pod use k8s pod graceful shutdown mechanism, user app is not even aware of this interface

**Comment by ramaraochavali** [comment]
> The long running pod use k8s pod graceful shutdown mechanism

That is handled today. Why do we need to terminate it forcefully? Sorry. I may not be understanding the intent.

**Comment by hzxuzhonghu** [comment]
`handleQuit` is for k8s job, it requires calling it explicitly before application returns. 

For pods created by deployment, user will no need to be aware of this api.

**Comment by hzxuzhonghu** [comment]
#15123

**Comment by ramaraochavali** [comment]
> Previously, with this job can exit successfully, but sometime later we added active connection check, which breaks the job container exit.

Active Connection check is not on by default. It needs to be enabled using this `EXIT_ON_ZERO_ACTIVE_CONNECTIONS` this metadata. And jobs can exit successfully as earlier by default now and that flag can be enabled only for regular sidecar apps.

**Comment by hzxuzhonghu** [comment]
Whatever it is a regression, and  active connection check can not work in most cases

**Pull request closed without merging**